airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/S3DestinationConfig.kt

```diff
@@ -26,11 +26,11 @@ import org.slf4j.LoggerFactory
  */
 open class S3DestinationConfig {
     @JvmField val endpoint: String?
-    @JvmField val bucketName: String
-    @JvmField val bucketPath: String
+    @JvmField val bucketName: String?
+    @JvmField val bucketPath: String?
     @JvmField val bucketRegion: String?
-    @JvmField val pathFormat: String
-    val s3CredentialConfig: S3CredentialConfig
+    @JvmField val pathFormat: String?
+    val s3CredentialConfig: S3CredentialConfig?
     @JvmField val formatConfig: S3FormatConfig?
```

Comment:
Want to convert these to property ? since they are accessed using getters. Or as a subsequent PR.